### PR TITLE
Add Image initialization expectations to YCbCrUpsampling

### PR DESCRIPTION
**Pull request by deymo**
Set the image initialization expectations after YCbCrUpsampling so we
can catch bugs right after it instead of relying on something else
further down using the uninitialized value.

This changes (x0, x1, y0, y1) so that all the pixels in that range are
defined.

### CODE REVIEW

File: lib/jxl/dec_reconstruct.cc [hunk 1]
```diff
@@ -274,17 +274,21 @@ void DoYCbCrUpsampling(size_t hs, size_t vs, ImageF* plane_in, const Rect& rect,
   // This X,Y range may include more pixels in the output than what we need.
   // Those pixels that fall outside the image boundary are undefined, but it is
   // safe to compute them with MulAdd().
```
[the remark this answers is not in the document]
Comment (deymo) [reply lib/jxl/dec_reconstruct.cc:276]: I wanted to get your opinion on that.

File: lib/jxl/dec_reconstruct.cc [hunk 2]
```diff
@@ -274,17 +274,21 @@ void DoYCbCrUpsampling(size_t hs, size_t vs, ImageF* plane_in, const Rect& rect,
...
-                       frame_dim.ysize_padded) -
-              frame_rect.y0();
-
-  size_t x0 = rect.x0() - lf.Padding();
-  size_t x1 = rect.x0() +
-              std::min(frame_rect.x0() + rect.xsize() + lf.Padding(),
-                       frame_dim.xsize_padded) -
-              frame_rect.x0();
+  const size_t y0 =
+      std::max<ssize_t>(frame_rect.y0() + rect.y0() - yoff - lf.Padding(), 0) +
```
Comment (veluca93) [inline lib/jxl/dec_reconstruct.cc:278]: I am reasonably sure this is wrong, and should instead be

`rect.y0() + std::max<ssize_t>(frame_rect.y0() - lf.Padding(), 0) - frame_rect.y0();`

File: lib/jxl/dec_reconstruct.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (veluca93) [reply lib/jxl/dec_reconstruct.cc:276]: I *think* it is not.

File: lib/jxl/dec_reconstruct.cc (hunk 2 above)
Comment (deymo) [reply lib/jxl/dec_reconstruct.cc]: I simplified that to `rect.y0() - std::min<size_t>(lf.Padding(), frame_rect.y0());`

**Review by veluca93 — APPROVED** [review]

**Pull request merged**